src/domain/session/room/timeline/tiles/BaseTextTile.js

```diff
@@ -56,4 +57,43 @@ export class BaseTextTile extends BaseMessageTile {
         }
         return this._messageBody;
     }
+
+    get replyPreviewBody() {
+        if (!this._entry.contextEventId) {
+            return null;
+        }
+        const entry = this._entry.contextEntry;
+        const error = this._generateError(entry);
+        if (error?.name === "ContextEntryNotFound") {
+            return { error };
+        }
+        const format = entry.content.format === "org.matrix.custom.html" ? BodyFormat.Html : BodyFormat.Plain;
+        const body = entry.content["formatted_body"] ?? entry.content["body"];
+        return {
```

Comment:
I'd create a `ReplyPreviewViewModel` (extending from ViewModel) class for this taking the entry and perhaps moving all the parsing logic in there, then you can also use getters for the avatar properties. Mentioned this before I think, but this will need to be cached like we already do for the main message body otherwise the binding would fire every time it is evaluated.